src/implementations/minecraft/configurable.rs

```diff
@@ -48,17 +60,148 @@ impl TConfigurable for Instance {
+        Supported(self.config.timeout_last_left)
+    }
+
+    fn timeout_no_activity(&self) -> MaybeUnsupported<Option<i32>> {
+        Supported(self.config.timeout_no_activity)
+    }
+
+    fn start_on_connection(&self) -> MaybeUnsupported<bool> {
+        Supported(self.config.start_on_connection)
+    }
+
+    fn backup_period(&self) -> MaybeUnsupported<Option<i32>> {
+        Supported(self.config.backup_period)
+    }
+
+    fn get_flavours(&self) -> Vec<String> {
+        vec![
+            "vanilla".to_string(),
+            "fabric".to_string(),
+            "paper".to_string(),
+        ]
+    }
+
+    fn set_name(&mut self, name: String) -> Result<(), traits::Error> {
         self.config.name = name;
+        self.write_config_to_file()?;
+        Ok(())
     }
 
-    fn set_description(&mut self, description: String) {
+    fn set_description(&mut self, description: String) -> Result<(), traits::Error> {
         self.config.description = description;
+        self.write_config_to_file()?;
+        Ok(())
     }
 
-    fn set_auto_start(&mut self, auto_start: bool) {
+    fn set_jvm_args(
+        &mut self,
+        jvm_args: Vec<String>,
+    ) -> MaybeUnsupported<Result<(), traits::Error>> {
+        self.config.jvm_args = jvm_args;
+        self.write_config_to_file()
+            .map_or_else(|e| Supported(Err(e)), |_| Supported(Ok(())))
+    }
+
+    fn set_min_ram(&mut self, min_ram: u32) -> MaybeUnsupported<Result<(), traits::Error>> {
+        self.config.min_ram = min_ram;
+        self.write_config_to_file()
+            .map_or_else(|e| Supported(Err(e)), |_| Supported(Ok(())))
+    }
+
+    fn set_max_ram(&mut self, max_ram: u32) -> MaybeUnsupported<Result<(), traits::Error>> {
+        self.config.min_ram = max_ram;
+        self.write_config_to_file()
+            .map_or_else(|e| Supported(Err(e)), |_| Supported(Ok(())))
+    }
+
+    fn set_auto_start(&mut self, auto_start: bool) -> MaybeUnsupported<Result<(), traits::Error>> {
         self.config.auto_start = auto_start;
-        *self.auto_start.lock().unwrap() = auto_start;
+        self.auto_start.store(auto_start, atomic::Ordering::Relaxed);
+        self.write_config_to_file()
+            .map_or_else(|e| Supported(Err(e)), |_| Supported(Ok(())))
+    }
+
+    fn set_restart_on_crash(
+        &mut self,
+        restart_on_crash: bool,
+    ) -> MaybeUnsupported<Result<(), traits::Error>> {
+        self.config.restart_on_crash = restart_on_crash;
+        self.auto_start
+            .store(restart_on_crash, atomic::Ordering::Relaxed);
+        self.write_config_to_file()
+            .map_or_else(|e| Supported(Err(e)), |_| Supported(Ok(())))
+    }
+
+    fn set_timeout_last_left(
+        &mut self,
+        timeout_last_left: Option<i32>,
+    ) -> MaybeUnsupported<Result<(), traits::Error>> {
+        match self.timeout_last_left.lock() {
+            Ok(mut v) => *v = timeout_last_left,
```

Comment:
unrelated to the functionality, idk what the `*` do, teach me why its needed afterward